keps/3589-manage-jobs-selectively/README.md

```diff
@@ -0,0 +1,232 @@
+rationale holds for `Deployment` and less strongly for `StatefulSet` that some mechanism
+to limit `manageJobsWithoutQueueName` is required otherwise it would be impossible in
+practice to ever set `manageJobsWithoutQueueName` to true.
+
+No other integrations beyond these three consider the `namespaceSelector` when deciding whether or not to
+suspend jobs that do not have a `queue-name` label.  This results in an irregular API and makes it
+harder to configure Kueue such that quotas will be enforced (cannot be bypassed by users).
+The irregularity between `batchv1/Job` and `Deployment` is likely to be especially confusing
+to users and cluster admins who would view both of these types as "built-in" to Kubernetes and
+would expect them to be configured in the same way.
+
+### Goals
+
+Provide a simple and uniform API to enable cluster admins to configure Kueue
+such that quotas will be enforced for specific namespaces.
+
+### Non-Goals
+
+Consider any scope for filtering `manageJobsWithoutQueueName` that is not based on namespaces.
+
+## Proposal
+
+We add a `managedJobsNamespaceSelector` of type `*metav1.LabelSelector`
+to the top level of the Kueue `Configuration` struct (same level as `manageJobsWithoutQueueName`).
+
+We use this new configuration for all integrations in the same way that the current
+`namespaceSelector` is used by the Pod integration.  Specifically,
+1. If `manageJobsWithoutQueueName` is false, `managedJobsNamespaceSelector` has no effect: Kueue will manage
+exactly those instances of supported Kinds that have a `queue-name` label.
+2. If `manageJobsWithoutQueueName` is true, then Kueue will (a) manage all instances of supported Kinds
+that have a `queue-name` label and (b) will manage all instances of supported Kinds that do not
+have a `queue-name` label if they are in namespaces that match `managedJobsNamespaceSelector`.
+
+We migrate away from using `podOptions.namespaceSelector` over the course of several releases.
+Ideally we remove it from the `Configuration` struct as part of going to the `v1` API level.
+
+### User Stories (Optional)
+
+#### Story 1
+
+Cluster admins will be able to use the combination of `managedJobsNamespaceSelector` and
+`manageJobsWithoutQueueName` to configure Kueue to enable quota enforcement on a per namespace basis.
+
+One sapproach would be to exempt selected namespaces from management:
+```yaml
+managedJobsNamespaceSelector:
+  matchExpressions:
+  - key: kubernetes.io/metadata.name
+    operator: NotIn
+    values: [ kube-system, kueue-system ]
+```
+Another approach would be for the cluster admin to label namespaces that are subject to management:
+```yaml
+managedJobsNamespaceSelector:
+  matchLabels:
+    kueue.x-k8s.io/managed-namespace: true
+```
+
+## Design Details
+
+### API
+
+The `Configuration` struct is extended to add ManageAllJobsNamespaceSelector
+```go
+type Configuration struct {
+
+  ...
+
+	// ManageJobsWithoutQueueName controls whether or not Kueue reconciles
```

Comment:
This is just a comment and not an objection to this feature.

I was wondering if we could just deprecate and remove the `manageJobsWithoutQueueName` and introduce more informable validations so that batch users can easily find the quota violation.

I think that the current manageJobsWithoutQueuenName does not have a good user experience since there is no way to find the reason why jobs are not runnable when they accidentally submit Jobs without queue names.

Because the Kueue automatically suspends the Jobs.

However, based on some consideration, this `manageJobsWithoutQueueName` has already been released. So, even if we do not implement this, we must follow the appropriate deprecation steps.

Additionally, if the cluster administrator provides the automatic queue-name insertion mechanism, this could be helpful for the batch users.
Furthermore, such an informable validation mechanism could be an alternative new feature since validation and mutation could not cover all use cases each other, and I think that both are valuable.